x-pack/plugins/security_solution/public/common/components/matrix_histogram/index.tsx

```diff
@@ -223,9 +242,21 @@ export const MatrixHistogramComponent: React.FC<MatrixHistogramComponentProps> =
             titleSize={titleSize}
             subtitle={subtitleWithCounts}
             inspectMultiple
+            showInspectButton={showInspectButton}
             isInspectDisabled={filterQuery === undefined}
           >
             <EuiFlexGroup alignItems="center" gutterSize="none">
+              <CasesContext owner={[APP_ID]} userCanCrud={userCanCrud ?? false}>
+                <VisualizationActions
```

Comment:
A couple of thoughts here:
- Wouldn't it make sense to move the `<CasesContext>` up to the _HostsContainer_ and the _NetworkContainer_ top-level components? So we will have cases available everywhere and we won't have to remember to wrap every single component that uses it? cc @stephmilovic

- The way we show the visualization actions button and the "_stackByOptions_" selector here seems a little bit odd to me, shouldn't the 3 dots button be the rightmost button? cc @monina-n 

![stack_3dots](https://user-images.githubusercontent.com/17747913/157427123-7a4662e2-daa6-411b-a1c3-2009ac4a7c76.png)

Reply:
I like @semd idea. I'd go as far as putting it all the way up for security solution as many other places use the CasesContext.

Reply:
yeah, I agree that it would be better to put it up to the HostsContainer and the NetworkContainer, but seems that `StatefulEventsViewer` on the page has been using `CasesContext` on its component level, moving `CasesContext` up to HostsContainer and the NetworkContainer level would wrap them again, so I am leaving it here. But I think we can create another issue to clean up the `CasesContext` for security solution.

Regarding to the position of `...`, I am open to any ideas, I am putting it on where inspect button was, if we would like to move it to the rightest corner, we might want to move inspect button as well.

Reply:
@angorayc you can put it up there and remove any inner one. You won't benefit from having two CasesContext inside each other and you don't risk breaking anything as long as you are sure you are wrapping what needs to be wrapped.

Reply:
I agree with @academo and @semd about the Cases context. I think as a lot of places in Security solution uses the `CasesContext` you can put it on the highest level possible. For example in the render of each sub plugin or even in the render of the whole security solution.

Reply:
@angorayc yeah it would be better to do it in a separate issue, we can remove all `CasesContext`s in the app and have just one at `x-pack/plugins/security_solution/public/app/index.tsx`.

About the `...` position, yes you're right, I didn't think about the normal inspect button. Maybe we should keep both in the left or move both to the right, or just one... IDK, let's wait for @monina-n

Reply:
I can get confirmation is we shold be using horizontal or vertical boxes for the menu icon

![Screen Shot 2022-03-09 at 10 49 44 AM](https://user-images.githubusercontent.com/67915896/157510874-970b4532-4c73-4004-a11e-0627d70c6301.png)

I'm ambivalent about placing tho conventions say it's typically the right-most item. We can discuss this at the explore sync, but I'm thinking if we should always have the three dots menu visible (instead of online on hover). As we now have a whole list of possible actions and the three dots menu is a very self-explanatory but not intrusive icon (instead of, for example, when we only had the inspect icon) and it will help with discoverability.

I feel like suggesting that might affect patterns in other parts of the app so let's talk about it at the sync

Reply:
Cleaning up CasesContext
https://github.com/elastic/kibana/issues/127335